csrc/scheduler/utils.cpp

```diff
@@ -1123,7 +1123,8 @@ std::vector<TensorView*> cacheInputs(Fusion* fusion, bool unroll) {
   auto in_tvs = ir_utils::filterByType<TensorView>(fusion->inputs());
   for (auto tv : in_tvs) {
     if (tv->uses().empty() || ir_utils::isTorchGatherLookupTv(tv) ||
-        ir_utils::isSelectInput(tv) || ir_utils::isIndexSelectLookupTv(tv)) {
+        ir_utils::isSelectInput(tv) || ir_utils::isIndexSelectLookupTv(tv) ||
+        ir_utils::isSliceInput(tv)) {
```

Comment:
Maybe we could make this a templated utility function like `ir_utils::isOpInput<Slice, PadOp, ...>(tv)`?

Reply:
(1) `isSelectInput` and `isSliceInput` can be merged into a variadic template function.
(2) `IndexSelectOp` and `TorchGatherOp`, they have two inputs: `lookupTv` and `indexTv`, but here we only check the `lookupTv`, if we merge them into the template then we need to add special code for these ops and someone may use it in a wrong way e.g. when checking for the `indexTv`.
so we keep `isIndexSelectLookupTv` and `isIndexSelectLookupTv`? or merge them into another variadic template function?